frontend/src/pages/modelRegistry/screens/RegisterModel/utils.ts

```diff
@@ -100,29 +163,100 @@ export const registerVersion = async (
+        // eslint-disable-next-line camelcase
+        string_value: formData.modelSourceName,
+      };
+    }
+
+    // For backward compatibility, set catalog custom properties
+    if (formData.catalogSourceName) {
+      artifactCustomProperties._registeredFromCatalogSourceName = {
+        metadataType: ModelRegistryMetadataType.STRING,
+        // eslint-disable-next-line camelcase
+        string_value: formData.catalogSourceName,
+      };
+    }
+
+    if (formData.catalogRepositoryName) {
+      artifactCustomProperties._registeredFromCatalogRepositoryName = {
+        metadataType: ModelRegistryMetadataType.STRING,
+        // eslint-disable-next-line camelcase
+        string_value: formData.catalogRepositoryName,
+      };
+    }
+
+    if (formData.catalogModelName) {
+      artifactCustomProperties._registeredFromCatalogModelName = {
+        metadataType: ModelRegistryMetadataType.STRING,
+        // eslint-disable-next-line camelcase
+        string_value: formData.catalogModelName,
+      };
+    }
+
+    if (formData.catalogModelTag) {
+      artifactCustomProperties._registeredFromCatalogTag = {
+        metadataType: ModelRegistryMetadataType.STRING,
+        // eslint-disable-next-line camelcase
+        string_value: formData.catalogModelTag,
+      };
+    }
+
+    const artifactData: CreateModelArtifactData = {
       name: `${formData.versionName}`,
-      description: formData.versionDescription,
-      customProperties: {},
+      description: formData.versionDescription || '',
+      customProperties: artifactCustomProperties,
       state: ModelArtifactState.LIVE,
       author,
       modelFormatName: formData.sourceModelFormat,
       modelFormatVersion: formData.sourceModelFormatVersion,
-      // TODO fill in the name of the data connection we used to prefill if we used one
-      // TODO this should be done as part of https://issues.redhat.com/browse/RHOAIENG-9914
-      // TODO should be fixed via https://issues.redhat.com/browse/RHOAIENG-19921
-      // storageKey: 'TODO',
+      // Set top-level pipeline source properties
+      modelSourceGroup: formData.modelSourceGroup,
+      modelSourceId: formData.modelSourceId,
+      modelSourceName: formData.modelSourceName,
+      // Set top-level catalog properties
+      catalogSourceName: formData.catalogSourceName,
+      catalogRepositoryName: formData.catalogRepositoryName,
+      catalogModelName: formData.catalogModelName,
+      catalogModelTag: formData.catalogModelTag,
       uri:
         formData.modelLocationType === ModelLocationType.ObjectStorage
           ? objectStorageFieldsToUri({
               endpoint: formData.modelLocationEndpoint,
               bucket: formData.modelLocationBucket,
               region: formData.modelLocationRegion,
               path: formData.modelLocationPath,
-            }) || '' // We'll only hit this case if required fields are empty strings, so form validation should catch it.
```

Comment:
I think this comment was helpful in explaining why we have the fallback `|| ''` since we usually want to avoid it but there's a valid reason here.